### DISCUSSION TITLE
Choosing an Item from Selection Does not Insert Text into Quill Editor

### DISCUSSION THREAD

**Discussion by pmaddaloni-shujinko** [afconsult/quill-mention#318] [Q&A]
I am not using the `quill` `formats` option. I'm really at a loss here. I'm using `react-quill v2.0.0-beta.2`. Maybe some incompatibility there? Is there a certain version of `quill` this targets?
![Kapture 2020-08-03 at 15 58 45](https://user-images.githubusercontent.com/56271286/89235031-465ade80-d5a2-11ea-8af7-fa8ff9c59421.gif)

Any help is appreciated!

```js

import 'quill-mention';
import PropTypes from 'prop-types';
import React, { useEffect, useRef, useState } from 'react';
import ReactQuill from 'react-quill';
import { renderToString } from 'react-dom/server';
import 'react-quill/dist/quill.snow.css';
import styles from './CommentForm.scss';
...

export default function CommentForm({

   ...

    const autocompleteModule = {
        mention: {
            isolateCharacter: true,
            source: function (searchTerm, renderList) {
                const matches = users.reduce((acc, user) => {
                    const { name: { firstName: first, lastName: last } } = user;
                    const name = `${first} ${last}`;
                    if (name.toLowerCase().includes(searchTerm.toLowerCase())) {
                        acc.push({ id: user.userId, value: name });
                    }
                    return acc;
                }, []);
                renderList(matches);
            },
            renderItem: function (item, searchTerm) {
                const [first, last] = item.value.split(' ');
                const avatar = renderToString(<UserAvatar firstName={first} lastName={last} />);
                return (`
                    <div>
                        ${avatar} <div>${item.value}</div>
                    </div>
                `);
            },
            mentionContainerClass: styles.mentionContainerClass,
            mentionListClass: styles.mentionListClass
        }
    };

...

   return(
        <form
            className={inlineEdit ? null : styles['comment-form']}
            method="post"
            onSubmit={onSubmit}
        >
                <ReactQuill
                    id={`Quill-${commentId}`}
                    theme="snow"
                    value={value}
                    onChange={onChange}
                    readOnly={readOnly}
                    modules={autocompleteModule}
                    {...rest}
                />

     ...
     </form>

}

**Comment by pmaddaloni-shujinko** [comment]
Turns out the `onChange` method that is passed to `ReactQuill` is being called with the wrong content (effectively the content prior to the `mentionDenotationChar`, e.g. `@`) , which gets the state of my component wrong which then overrides the insertion in the `quill` editor. Any idea how to stop the onChange from being called with the wrong content?

**Comment by fridaystreet** [comment]
not sure if it's related but I have the reverse. I can select anoption with the mouse, but I can't press enter when one is highlighted. it just closes the selections and goes back to editor with the cursor in front of the @

erro in console is addRange(): The given range isn't in document.

Now I've had that error with other plugins and it was because the editor was getting re-rendered. Once I removed all the annonymous functions in my components related to the editor, this stopped happening. not sure if that is related to this or not

**Comment by shamoons** [comment]
Anyone find a resolution to this?

**Comment by Shubhra1028** [comment]
Any solution to this?

**Comment by artknight** [comment]
I have the same issue!!! Running Quill v2.0.0-dev.4. 

Does ANYONE have a solution for this?

**Comment by artknight** [comment]
I have fixed all the issues related to Quill v2.0.0-dev.4. Here is the build. https://github.com/artknight/quill-mention/tree/master/dist

Feel free to ping me if anything :)
